### DISCUSSION TITLE
technical problems

### DISCUSSION THREAD

**Discussion by mithrendal** [vAmigaWeb/vAmigaWeb#2] [General]
This discussion is to have an exchange on technical problems on a variety of things which we encounter when doing web ports. Problems could for example be browser, SDL or emsdk related ...

**Comment by mithrendal** [comment]
Ok I will be first... @chris70c already mentioned that he faced some difficulties with DOM event listeners that are added by emscripten ... he said that he did patched to gt them out of the way...

I now faced also problems when making vAmigaWeb with these event listeners which I don't really need, as I do everything in pure Javascript.

I did a small patch to make it work for now ... I added this patch to the emscripten shell.html template 


      //patch to move problematic event listeners out of the way
      setTimeout(()=>{
        window.registerMouseEventCallback=function(){};
        window.listenOnce=function(){};
      },3000);

**Comment by chris70c** [comment]
This is my patched SDL file if you want to try it out, it removes all listeners added by default.
[mod_SDL_emscriptenevents.zip](https://github.com/vAmigaWeb/vAmigaWeb/files/7317535/mod_SDL_emscriptenevents.zip)

Dirk here's the test game for the sid switching issue, press 2 for the intro and it will be very noticeable what the problem is changing sid engine
[Soulless_+6D_LAXITY.zip](https://github.com/vAmigaWeb/vAmigaWeb/files/7317538/Soulless_%2B6D_LAXITY.zip)
.

**Comment by mithrendal** [comment]
I just learned that I should have opened this discussion thread under teams ... 🙄 is there a way to transfer?
